# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

<div class="span5 alert alert-info">
### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
</div>
****

In [21]:
import pandas as pd
import numpy as np
import math
from scipy import stats

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding names
# selects only the race rows with 'b' and sums the calls
sum(data[data.race=='b'].call)

157.0

In [4]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [11]:
data_b = data[data.race=='b']
data_w = data[data.race=='w']

In [14]:
len(data_w)

2435

In [15]:
len(data_b)

2435

# Question 1:
What test is appropriate for this problem?

This problem requires a comparison of mean values based on call backs.  If the mean values of the two samples are statistically different then we can assume that the factor of racialized names plays a significant role.

Assumptions:
 - set population mean to sample mean as the sample set is sufficiently large @ 2435 rows (CLT will hold)
 - use a z statistic and compute the p value

# Question 2: The Hypotheses
Ho: That the average callbacks of black sounding names is equal to the average callbacks of white sounding names

Ha: That the callback rate is not equal

    - Ha: mu1 is not equal to mu2
    
Let m represent the black names and n the white
Let 1 represent the black names and 2 the white

Two Tailed Z Test for inequality of means
Assume an alpha level of 0.05 so critical value is +/-1.67

Test Statistic: Z = (Xbar -Ybar -(mu1-mu2))/(sqrt(std1^2/m + std2^2/n)

# Question 3: MoE, CI and P-value


In [29]:
b_sample_std = data_b['call'].std()
w_sample_std = data_w['call'].std()
MoE_b = 1.67*b_sample_std
MoE_w = 1.67*w_sample_std
MoE_b-MoE_w

-0.08299241691827774

Confidence Interval for mu1-mu2

Xbar-Ybar +/- Z(0.025)*sqrt(std1^2/m + std2^2/n)

Which computes as:

In [30]:
confidence_bound=1.67*math.sqrt(b_sample_std/2435+w_sample_std/2435)

In [31]:
data_b['call'].mean()-data_w['call'].mean() + confidence_bound

-0.007140636454251659

In [32]:
data_b['call'].mean()-data_w['call'].mean() - confidence_bound

-0.0569250732558695

That is to say with 95% confidence -0.057 < mu1 - mu2 < -0.007

In [34]:
Z = (data_b['call'].mean()-data_w['call'].mean())/math.sqrt(b_sample_std/2435+w_sample_std/2435)
Z

-2.149059868694256

P-value = 2*(1- omega(abs(Z)))= 1-0.9838 = 0.0324

# Question 4
Since -2.15 is not within the range of +/-1.67 we may reject the null hypothesis that the means are equal

This is supported by a P-value of 0.0324 which is to say that we can reject Ho at the value of alpha = 0.05

The data supports the opinion that racialized names play a significant role in determining call backs.

# Question 5
It is not possible with this level of analysis to determine that race is the most important factor in determining call backs.  A multivariate analysis would be required to compare the effect of each feature